# HG changeset patch
# User Rik <octave@nomad.inbox5.com>
# Date 1276620900 25200
#      Tue Jun 15 09:55:00 2010 -0700
# Node ID 6f34e781aed2e9047bcc1c741330d565bc107eef
# Parent  d500acdcf2dde0dc7dab137488c1eb47a21ab7d8
Second try at removing extra newlines from NDArrays

diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,8 +1,12 @@
+2010-06-15  Rik <octave@nomad.inbox5.com>
+
+	* pr-output.cc: Second try at removing extra newlines from NDArrays
+
 2010-06-14  Rik <octave@nomad.inbox5.com>
 
 	* pr-output.cc: Remove extra newlines when displaying multi-dimensional
         arrays (bug #30029).
 
 2010-06-14  Jaroslav Hajek  <highegg@gmail.com>
 
 	* data.cc (Fsort, Fnth_element): Fix behavior when trailing singleton
diff --git a/src/pr-output.cc b/src/pr-output.cc
--- a/src/pr-output.cc
+++ b/src/pr-output.cc
@@ -1890,24 +1890,24 @@ octave_print_internal (std::ostream& os,
               idx(1) = idx_vector (':'); \
  \
               for (int k = 2; k < ndims; k++) \
                 idx(k) = idx_vector (ra_idx(k)); \
  \
               octave_value page \
                 = MAT_T (Array<ELT_T> (nda.index (idx), nr, nc)); \
  \
-              if (i < ndims - 2) \
+              if (i != m - 1) \
                 { \
                   page.print_with_name (os, nm); \
                 } \
               else \
                 { \
                   page.print_name_tag (os, nm); \
-                  page.print_raw(os); \
+                  page.print_raw (os); \
                 } \
               \
               if (i < m) \
                 NDA_T::increment_index (ra_idx, dims, 2); \
             } \
         } \
     } \
   while (0)
